### PR TITLE
Add API reference for getAssetFromKV

### LINKED ISSUE #48: include interface in readme
this package assumes (generally) that it will be used in conjunction with wrangler using the `[site]` feature, but it would be useful to include in the readme more about the different options one can pass to the various functions, as well as the assumptions about bindings present in the worker.

### PR DESCRIPTION
**Pull request by ashleymichal**
Closes #48 

Adds description and explanation of all configuration options for the function `getAssetFromKV`. Also adds argument/return values for other helper functions supplied by package.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,9 +47,45 @@ async function handleEvent(event) {
 }
 ```
 
+### Options
+
+This function was designed to work with [Worker Sites](TODO:link-marketing-page), but you can customize the behavior of this function by passing the following options into the second argument to `getAssetFromKV`.
+
+#### `ASSET_NAMESPACE`
+
+type: KV Namespace Binding (required if not using Wrangler --site feature)
```
Comment (victoriabernard92) [inline README.md]:
```suggestion
type: KV Namespace Binding
```

File: README.md [hunk 2]
```diff
@@ -54,9 +47,45 @@ async function handleEvent(event) {
...
+
+We assume that your Worker includes a [binding to a Workers KV Namespace](TODO:link-documentation) named `__STATIC_CONTENT`. This binding can be configured by passing the `ASSET_NAMESPACE` option to `getAssetFromKV`. It is further assumed that this namespace consists of static assets such as html, css, javascript, or image files, keyed off of a relative path that roughly matches the assumed url pathname of the incoming request.
+
+#### `__STATIC_CONTENT_MANIFEST` (optional)
```
Comment (victoriabernard92) [inline README.md]: I think we should split off into two sections,  the internal-only (internal to the package) vars and the ones we expect users to use in their index.js.

File: README.md [hunk 3]
```diff
@@ -54,9 +47,45 @@ async function handleEvent(event) {
...
+By default, mapRequestToAsset takes any path that ends in `/` or evaluates to an html file and appends `index.html` or `/index.html` for lookup in your Workers KV namespace. This works for most static site generators, but you can customize this behavior by passing your own function as `mapRequestToAsset`. The function should take a `Request` object as its only argument, and return a string representation of the path to be looked up in the asset manifest/KV.
+
+#### `cacheControl`
+
+type: object
+
+`cacheControl` allows you to configure options for both the Cloudflare Cache accessed by your Worker, and the browser cache headers sent along with your Workers' responses. The default values are as follows:
+
+```javascript
+let cacheControl = {
```
Comment (victoriabernard92) [inline README.md]: If we are going to add this we should define each property

File: README.md (hunk 1 above)
Comment (victoriabernard92) [reply README.md]: Adding this here adds confusion IMO. I think we should clearly define what each member does and then put an \* next to the ones that Workers Sites will configure for you. I can PR to this branch to show what I mean if you'd find that helpful @ashleymichal

File: README.md (hunk 3 above)
Comment (victoriabernard92) [reply README.md]: e.g. 
```
##### bypassCache
type: number

Sets `Cache-Control` header `max-age` to the number passed in (in seconds) to the response which will be used by browser/client caches. X days by default.

File: README.md (hunk 1 above)
Comment (ashleymichal) [reply README.md]: hmm. it is confusing. interested in what that looks like yah.

File: README.md (hunk 3 above)
Comment (ashleymichal) [reply README.md]: i mean, we have to add it, it's part of the API. so yah we have to add that

File: README.md (hunk 2 above)
Comment (ashleymichal) [reply README.md]: the presentation is these as options; they all have defaults, and this explains what they are, but if folks are using any of them, they need to pass them in at the call site.

File: README.md (hunk 3 above)
Comment (ashleymichal) [reply README.md]: added. lmk what you think.

File: README.md [hunk 4]
```diff
@@ -11,17 +11,12 @@ npm i @cloudflare/kv-asset-handler
 
 ## Usage
 
-### `getAssetFromKV`
-
-`getAssetFromKV` that maps `FetchEvent` objects to KV Assets, and throws an `Error` if it cannot.
+This package was designed to work with [Worker Sites](TODO:link-marketing-page).
```
Comment (ashleymichal) [inline README.md]:
```suggestion
This package was designed to work with [Worker Sites](https://workers.cloudflare.com/sites).
```

File: README.md [hunk 5]
```diff
@@ -82,6 +127,27 @@ async function handleEvent(event) {
 }
 ```
 
-### `mapRequestToAsset`
+#### `ASSET_NAMESPACE`
+
+type: KV Namespace Binding
+
+The binding name to the KV Namespace populated with key/value entries of files for the Worker to serve. By default, Workers Sites uses a [binding to a Workers KV Namespace](TODO:link-documentation) named `__STATIC_CONTENT`.
```
Comment (ashleymichal) [inline README.md]:
```suggestion
The binding name to the KV Namespace populated with key/value entries of files for the Worker to serve. By default, Workers Sites uses a [binding to a Workers KV Namespace](https://developers.cloudflare.com/workers/reference/storage/api/#namespaces) named `__STATIC_CONTENT`. 
```

File: README.md [hunk 6]
```diff
@@ -82,6 +127,27 @@ async function handleEvent(event) {
...
+return getAssetFromKV(event, { ASSET_NAMESPACE: MY_NAMESPACE })
+```
+ 
+#### `ASSET_MANIFEST` (optional)
+
+type: text blob (JSON formatted)
+
+The mapping of requested file path to the key stored on Cloudflare.
+
+Workers Sites with Wrangler bundles up a text blob that maps request paths to content-hashed keys that are generated by Wrangler as a [cache-busting measure](TODO:link-sites-blog-post). If this option/binding is not present, the function will fallback to using the raw pathname to look up your asset in KV. If, for whatever reason, you have rolled your own implementation of this, you can include your own by passing a stringified JSON object where the keys are expected paths, and the values are the expected keys in your KV namespace.
```
Comment (ashleymichal) [inline README.md]:
```suggestion
Workers Sites with Wrangler bundles up a text blob that maps request paths to content-hashed keys that are generated by Wrangler as a [cache-busting measure](https://blog.cloudflare.com/extending-the-workers-platform/). If this option/binding is not present, the function will fallback to using the raw pathname to look up your asset in KV. If, for whatever reason, you have rolled your own implementation of this, you can include your own by passing a stringified JSON object where the keys are expected paths, and the values are the expected keys in your KV namespace.
```

File: README.md [hunk 7]
```diff
@@ -54,9 +49,59 @@ async function handleEvent(event) {
...
+
+You can customize the behavior of `getAssetFromKV` by passing the following properties as an object into the second argument 
+
+```
+return getAssetFromKV(event, { mapRequestToAsset: ... })
+```
+
+#### `mapRequestToAsset`
+
+type: function(Request) => Request
```
Comment (victoriabernard92) [inline README.md]:
```suggestion
type: (req: Request) => Request
```
nit TS syntax

File: README.md [hunk 8]
```diff
@@ -54,9 +49,59 @@ async function handleEvent(event) {
...
+
+Sets the `Cache-Control: max-age` header on the response used as the edge cache key. By default set to 2 days (`2 * 60 * 60 * 24`).
+
+##### `bypassCache`
+
+type: boolean
+
+Determines whether to cache requests on Cloudflare's edge cache. By default set to `false` (recommended for production builds). Useful for development when you need to eliminate the cache's effect on testing.
+
 ### `serveSinglePageApp`
```
Comment (victoriabernard92) [inline README.md]:
```suggestion
### `serveSinglePageApp`
type: () => void
```

File: README.md (hunk 7 above)
Comment (ashleymichal) [reply README.md]: is not TS yet

Comment (victoriabernard92) [reply README.md]: just more proper for any JS project ( I had it wrong at first too) just a nit.

Comment (ashleymichal) [reply README.md]: hrm. is this even a, like, "normal" documentation syntax?

File: README.md [hunk 9]
```diff
@@ -54,9 +47,45 @@ async function handleEvent(event) {
 }
 ```
 
+### Options
+
+This function was designed to work with [Worker Sites](TODO:link-marketing-page), but you can customize the behavior of this function by passing the following options into the second argument to `getAssetFromKV`.
```
[the remark this answers is not in the document]
Comment (ashleymichal) [reply README.md]: i think i did this.

File: README.md [hunk 10]
```diff
@@ -54,9 +49,60 @@ async function handleEvent(event) {
...
+Sets the `Cache-Control: max-age` header on the response used as the edge cache key. By default set to 2 days (`2 * 60 * 60 * 24`).
+
+##### `bypassCache`
+
+type: boolean
+
+Determines whether to cache requests on Cloudflare's edge cache. By default set to `false` (recommended for production builds). Useful for development when you need to eliminate the cache's effect on testing.
+
 ### `serveSinglePageApp`
+type: () => void
```
Comment (ashleymichal) [inline README.md]:
```suggestion
type: function() => function(request) => string
```

File: README.md [hunk 11]
```diff
@@ -54,34 +49,117 @@ async function handleEvent(event) {
...
+
+## Other functions
+
+#### `mapRequestToAsset`
+
+type: function(Request) => Request
+
+The default function for mapping incoming requests to keys in Cloudflare's KV.
+
+Takes any path that ends in `/` or evaluates to an html file and appends `index.html` or `/index.html` for lookup in your Workers KV namespace. This works for most static site generators, but you can customize this behavior by passing your own function as `mapRequestToAsset`. The function should take a `Request` object as its only argument, and return a string representation of the path to be looked up in the asset manifest/KV.
```
Comment (ashleymichal) [inline README.md]:
```suggestion
Takes any path that ends in `/` or evaluates to an html file and appends `index.html` or `/index.html` for lookup in your Workers KV namespace.
```

File: README.md [hunk 12]
```diff
@@ -54,34 +49,117 @@ async function handleEvent(event) {
...
+getAssetFromKV(event, { mapRequestToAsset: ... })
+```
+
+#### `mapRequestToAsset`
+
+type: function(Request) => Request
+
+Maps the incoming request to the value that will be looked up in Cloudflare's KV
+
+By default, mapRequestToAsset is set to the exported function [`mapRequestToAsset`](#maprequesttoasset-1)
```
Comment (ashleymichal) [inline README.md]:
```suggestion
By default, mapRequestToAsset is set to the exported function [`mapRequestToAsset`](#maprequesttoasset-1).  This works for most static site generators, but you can customize this behavior by passing your own function as `mapRequestToAsset`. The function should take a `Request` object as its only argument, and return a string representation of the path to be looked up in the asset manifest/KV.
```

File: README.md [hunk 13]
```diff
@@ -54,34 +49,117 @@ async function handleEvent(event) {
...
+getAssetFromKV(event, { mapRequestToAsset: ... })
+```
+
+#### `mapRequestToAsset`
+
+type: function(Request) => Request
+
+Maps the incoming request to the value that will be looked up in Cloudflare's KV
+
+By default, mapRequestToAsset is set to the exported function [`mapRequestToAsset`](#maprequesttoasset-1).  This works for most static site generators, but you can customize this behavior by passing your own function as `mapRequestToAsset`. The function should take a `Request` object as its only argument, and return a string representation of the path to be looked up in the asset manifest/KV.
```
Comment (ashleymichal) [inline README.md]:
```suggestion
By default, mapRequestToAsset is set to the exported function [`mapRequestToAsset`](#maprequesttoasset-1).  This works for most static site generators, but you can customize this behavior by passing your own function as `mapRequestToAsset`. The function should take a `Request` object as its only argument, and return a new `Request` object with an updated path to be looked up in the asset manifest/KV.
```

**Pull request merged**